etcd/v1alpha1/types_pacemakercluster.go

```diff
@@ -0,0 +1,516 @@
+	// QuorumStatusType can be one of the following values:
+	// - Quorate - the cluster has quorum
+	// - NoQuorum - the cluster does not have quorum
+	// When present, it must be a valid QuorumStatusType.
+	// When not present, the quorum status is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +optional
+	QuorumStatus QuorumStatusType `json:"quorumStatus,omitempty"`
+
+	// nodesOnline is the count of online nodes
+	// When present, it must be a valid integer between 0 and 2 inclusive.
+	// When not present, the nodes online count is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=2
+	// +optional
+	NodesOnline *int32 `json:"nodesOnline,omitempty"`
+
+	// nodesTotal is the total count of configured nodes
+	// When present, it must be a valid integer between 0 and 2 inclusive.
+	// When not present, the nodes total count is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=2
+	// +optional
+	NodesTotal *int32 `json:"nodesTotal,omitempty"`
+
+	// resourcesStarted is the count of started resources
+	// When present, it must be a valid integer between 0 and 16 inclusive. For a healthy Two Node Fencing (TNF) cluster, this is expected to be 6.
+	// The expected resources are kubelet, etcd, and a fencing resource (i.e. redfish) for each node.
+	// The number could be less than 6 if the cluster is starting up or not healthy.
+	// The total number of resources managed by the cluster could drift over time as Two Node Fencing matures, so this is left flexible.
+	// When not present, the resources started count is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=16
+	// +optional
+	ResourcesStarted *int32 `json:"resourcesStarted,omitempty"`
+
+	// resourcesTotal is the total count of configured resources
+	// When present, it must be a valid integer between 0 and 16 inclusive. For a healthy Two Node Fencing (TNF) cluster, this is expected to be 6.
+	// The expected resources are kubelet, etcd, and a fencing resource (i.e. redfish) for each node.
+	// The total number of resources managed by the cluster could drift over time as Two Node Fencing matures, so this is left flexible.
+	// When not present, the resources total count is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=16
+	// +optional
+	ResourcesTotal *int32 `json:"resourcesTotal,omitempty"`
+}
+
+// NodeStatus represents the status of a single node in the Pacemaker cluster
+type PacemakerNodeStatus struct {
+	// name is the name of the node
+	// It must be a valid string between 1 and 256 characters long and cannot be empty.
```

Comment:
Is the name tied at all to the Kubernetes Node name?

Reply:
Yes - is there special validation for k8s node names I can invoke here? added it to the doc